From 30ea9d422b6e04e7e4f145696196df7d9cda3d88 Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Tue, 5 Sep 2023 15:53:29 +0530
Subject: [PATCH] Fix_For_Set_systemboot_option 0

Description:
Set system boot option 0,Setting Reserve the value,it will setting.instead of It sholud throw the error.So,Add the condition to check the reserve value.

Tested:
Build Successfully and Verified,its Working As Expected.

Before Fix:
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.100.168 -U root -P 0penBmc -I lanplus -C 17 raw 0x00 0x08 0x00 0x03

msgubuntu4@msgubuntu4-OptiPlex-5080:

After Fix:
msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.99.188 -U root -P 0penBmc -I lanplus -C 17 raw 0x00 0x08 0x00 0x03
Unable to send RAW command (channel=0x0 netfn=0x0 lun=0x0 cmd=0x8 rsp=0xcc): Invalid data field in request

Working As Expected.

Signed-off-by: saravananm <saravananm@ami.com>
---
 chassishandler.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/chassishandler.cpp b/chassishandler.cpp
index 6d80562..336cfd9 100644
--- a/chassishandler.cpp
+++ b/chassishandler.cpp
@@ -2037,10 +2037,16 @@ ipmi::RspType<> ipmiChassisSetSysBootOptions(ipmi::Context::ptr ctx,
     {
         uint2_t setInProgressFlag;
         uint6_t rsvd;
+	constexpr uint8_t setInProgressMask = 0x02;
+
         if (data.unpack(setInProgressFlag, rsvd) != 0 || !data.fullyUnpacked())
         {
             return ipmi::responseReqDataLenInvalid();
         }
+	if (setInProgressFlag > setInProgressMask)
+        {
+            return ipmi::responseInvalidFieldRequest();
+        }
         if (rsvd)
         {
             return ipmi::responseInvalidFieldRequest();
-- 
2.25.1

